frontend/src/components/Gym/ConfirmationModal.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import Button from '../Button'
+import styled from 'styled-components'
+
+const OverlayContainer = styled.div<{ overlayBackgroundColor: string }>`
+  position: fixed;
+  min-height: 100vh;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: ${(props) => (props.overlayBackgroundColor ? props.overlayBackgroundColor : 'rgba(0, 0, 0, 0.3)')};
+  z-index: 999;
+`
+
+const MainContainer = styled.div`
+  position: fixed;
+  background-color: #fff;
+  width: 90vw;
+  border-radius: 15px;
+  margin: auto;
+  padding: 15px;
+  box-shadow: 0 2px 5px 1px #888888;
+  z-index: 1000;
+  display: flex;
+  flex-direction: column;
+`
+
+const TitleContainer = styled.div`
+  text-align: center;
+`
+
+const TitleText = styled.h1`
+  font-family: Lato;
+  font-size: 16px;
+  font-weight: 400;
+`
+
+const DescriptionText = styled.p`
+  font-weight: 250;
+  font-size: 14px;
+  padding-top: 0.5rem;
+  color: #33363a;
+  text-align: justify;
+`
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-evenly;
+  width: 100%;
+  margin-top: 10px;
+`
+
+type Props = {
+  title: string
+  description?: string | JSX.Element
+  // Left Button Props
+  leftButtonType: 'primary' | 'secondary'
+  leftButtonText: string
+  onLeftButtonClick: React.MouseEventHandler<HTMLButtonElement>
```

Comment:
maybe making the left button optional would be better? since usually it is used to close the modal, so the default could be 
leftButtonType = 'secondary'
leftButtonText = 'Cancel'
onLeftButtonClick = close modal
and they will only be customised if its specified